libcommon/securestr.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * securestr.c - Make strings safe for printing and beat CVE-2024-28085.
+ */
+/*
+ * Copyright (C) 2024: Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ *
+ * Sections of the CVE-2024-28085 mitigation borrowed from 4.3BSD
+ * Net/2's cat(1) implementation for the '-v' option. It differs
+ * from Heirloom native/UNIX 7th's cat(1) per not being so dependent
+ * on magic numbers and using functions from the "isx()" family.
+ * As per the copyright header of Net/2's "bin/cat/cat.c":
+ * Copyright (c) 1989 The Regents of the University of California.
+ * All rights reserved.
+ *
+ * This code is derived from software contributed to Berkeley by
+ * Kevin Fall.
+ *
+ * SPDX-Licence-Identifier: BSD-4-Clause-UC
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+char *ssafe(char *s) {
+	int i = 0,
+	    j = 0;
+	char safe[BUFSIZ] = "",
+	     *ssafe = "";
+
+	for (i = 0; s[i]; i++) {
```

Comment:
You could just advance `s` instead of using `i`

Reply:
I think this is far more legible and it works just the same.

Reply:
> I think this is far more legible and it works just the same.

Ok but why is `i` incremented on more than one place?

Reply:
That's some alchemy with the string when escaping control characters. I've talked about it [here](https://github.com/Projeto-Pindorama/heirloom-ng/pull/59#discussion_r1794618920) and there's nothing we can do further on these.

Reply:
Closed for being outdated